torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -617,13 +672,16 @@ def _pre_load_state_dict_hook(
     if torch.cuda.is_available():
         torch.cuda.synchronize()
     # Dispatch into state_dict specific implementation of pre-hook.
-    _pre_load_state_dict_hook_fn[fsdp_module._state_dict_type](
-        fsdp_module, state_dict, prefix
+    _pre_load_state_dict_hook_fn[fsdp_state._state_dict_type](
+        fsdp_module, fsdp_state, state_dict, prefix
     )
 
 
+@no_type_check
 @torch.no_grad()
 def _post_load_state_dict_hook(module: nn.Module, *args: Any) -> None:
+    # TODO: get the composable state from module
```

Comment:
Just FYI, I think the syntax is like:
```
fsdp_state = fully_shard.state(module)
```

For example:
https://github.com/pytorch/pytorch/blob/fca6ed02b91e0d685cc9b8b504bac5d356d31876/torch/distributed/_composable/fully_shard.py#L44

Reply:
hmm, this can cause a circular import. Can we pull this to a module that does not depend on FSDP?

Reply:
Discussed over chat -- let us discuss this further with the composability devs